### PR TITLE
Print line breaking after file path with errors

### LINKED ISSUE #14148: --list-different logs errors after file path instead of in a new line
**Environments:**

- Prettier Version: 2.8.2
- Running Prettier via: CLI
- Operating System: all

**Steps to reproduce:**

create a index.js file with a syntax issue in it, then run:
```bash
npx prettier --list-different "./**/*.{js,jsx,ts,tsx,scss,json}"
```

**Actual behavior:**
the output will be:
```log
path/to/file/index.js[error] path/to/file/index.js: SyntaxError: Unexpected token, expected "," (5:1)
[error]   3 |
[error]   4 | console.log("test: create a js syntax error here"
[error] > 5 | export { test };
[error]     | ^
[error]   6 |
```

**Expected behavior:**
```log
path/to/file/index.js
[error] path/to/file/index.js: SyntaxError: Unexpected token, expected "," (5:1)
[error]   3 |
[error]   4 | console.log("test: create a js syntax error here"
[error] > 5 | export { test };
[error]     | ^
[error]   6 |
```
Also, it would be great to have a flag (e.g. `--no-error`) that will stop prettier from logging errors, and it will just log the file path, so the output will be:
```log
path/to/file/index.js
```

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Fixes #14148

Previously, only the --write option printed a newline before the error, but other options and no options print a newline as well.

### before

```console
test.js[error] test.js: SyntaxError: Unexpected token: ')' (1:6)
[error] > 1 | 1 (+-) hoge
[error]     |      ^
```

### after

```console
test.js
[error] test.js: SyntaxError: Unexpected token: ')' (1:6)
[error] > 1 | 1 (+-) hoge
[error]     |
```

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by fisker — APPROVED** [review]

File: src/cli/format.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ function handleError(context, filename, error, printedFilename) {
     return;
   }
 
-  if (context.argv.write) {
+  if (printedFilename) {
     // Add newline to split errors from filename line.
     process.stdout.write("\n");
   }
```
Comment (fisker) [inline src/cli/format.js:44]:
```
>node bin/prettier .editorconfig
.editorconfig[error] No parser could be inferred for file "E:\project\@forks\prettier\.editorconfig".
```

Comment (fisker) [reply src/cli/format.js:44]: Maybe like this on top? `printedFilename` must be clear or add a new line.

```js
if (printedFilename) {
    if ((context.argv.write || context.argv.ignoreUnknown) && (error instanceof errors.UndefinedParserError)) {
      printedFilename.clear();
    } else {
      process.stdout.write("\n");
    }
}
```

**Comment by fisker** [comment]
Let's use `.unknown` as file extension.

**Comment by sosukesuzuki** [comment]
done

**Review by fisker — APPROVED** [review]
Looks good

**Pull request merged**